### DISCUSSION TITLE
Discrepancy in dexcom g6

### DISCUSSION THREAD

**Discussion by Billibilli** [NightscoutFoundation/xDrip#2301] [Q&A]
Hi there, 

First time Dexcom G6 & Xdrip user. Transmitter serial start with 8E* . Xdrip version e557cba

Xdrip/Dexcom has been giving rather weird reading, it is already 24 hours (a few hours later it will be 48 hours) from the time it activated. It seems always there is 1.6-1.8 mmol/L difference now. I haven't compared this with Dexcom receiver because I have none. And my cell phone is huawei, which couldn't install google play therefore not possible to use the Dexcom official software. So I couldn't say 100% it is Xdrip's fault, but my feeling of the error pattern is rather like software instead of hardware. **Anyway I would like to know what setting can I check to ensure maximal accuracy from Xdrip ?If calibration is required, what is the recommended procedure to do it in Xdrip?** 

![d4d94037d253e74e1d72fc60346b187](https://user-images.githubusercontent.com/4047002/187030487-d849b434-7cde-4349-8900-0966b9194c7c.jpg)

![c4358742c23d923373cc494ba843a3b](https://user-images.githubusercontent.com/4047002/187030507-d2444ee8-b487-410c-85f4-b95fed6951b9.jpg)

The later is FreeStyle Libre 1, which has been verified with blood test multiple times, the value are accurate, leaving the Xdrip/Dexcom not accurate. 

There was a long signal loss possibly due to the cell phone operating system shutting down the software. After that event happened, the discrepancy are doubled (previously the discrepancy is 0.9, now it is 1.8) . 

![3e99e0ae782426a41a275a9917083f1](https://user-images.githubusercontent.com/4047002/187030756-fadb24d6-d7fe-4e30-bb88-d5e65d1acefd.jpg)

The system status and log doesn't seem to be wrong too much. However, the resistance seems to be weird in the log, I don't know what that term means but 0 doesn't seem like a legit number. 

![753e7cd86b3f5c9ea979706416f128c](https://user-images.githubusercontent.com/4047002/187030876-1b6e483e-003b-4411-9d26-e0d41bf7649a.jpg)

![1661606456(1)](https://user-images.githubusercontent.com/4047002/187032107-f0650659-d23c-46ef-a985-00a6102eeac0.png)

Settings (Not the same with [G6 Recommended Settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html)): 
G5/G6 debug setting  : 

* Use the OB1 Collector (on)
* Native Algorithim (on)
* Minimal Scanning(on)
* Avoid Scanning(on)
* Go on without calibrations (on)
* Allow OB1 initiate bonding (on)
* **Allow OB1 unbonding (on)**
* G6 Support (on)
* Authenticate G5 before each reading (on)

Bluetooth Setting: 

* Turn Bluetooth (on)
* Bluetooth Watchdog (on)
* **Timer value for BT watchdog: 60min** (on)
* G5 Bluetooth Watchdog (on)
* Trust Auto-Connect (on)
* Use Background Scans (on)
* Allow bluecon unbounding (on)

Misc Setting: 
* Samsung workaround  (on)
* Run Collector in the foreground (on)

~~Trying to understand the code, I found there is a functions seems changing the raw data from Dexcom G6 before a certain timing.~~

https://github.com/NightscoutFoundation/xDrip/blob/984ac395a400a3f46b289fa53c646975c522d244/wear/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java#L1695-L1703

~~Is there anyone who can explain why this is neccesary? also how can i see the log info in the app? is there a log filter option/developer debug option I can turn on to see this? I think from the logic, this has to be printed into the log, yet i couldn't search find that in the error and event list.~~

EDITED: 

Never mind I found another discuission on this https://github.com/NightscoutFoundation/xDrip/discussions/2137

EDITED: 

There are some confusions I have on the help documents.

1) 
[Starting Dexcom G6](https://navid200.github.io/xDrip/docs/Starting-G6.html)
> A common mistake is to replace an old transmitter with a new one, stop sensor, change transmitter ID, and start sensor right away. This could result in the new transmitter never receiving the start command at the right time. The outcome is no readings after the 2-hour warm-up.

So what should be done instead? What are the correct routine to do in Xdrip if a) replace the sensor only. b) replace the transmitter and the sensor? 

2) 
[G6 Recommended Settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html)

> Disable Settings −> Less common settings −> Advanced Calibration −> Automatic Calibration

What does this do? 

3) 

Is there a conflict between _Allow OB1 unbonding_ and those controling Bluetooth on/off setting in bluetooth setting such as _G5 Bluetooth Watchdog_? If I check _G5 Bluetooth Watchdog_, does it mean I should also check _Allow OB1 unbonding_ ?

**Comment by Navid200** [comment]
I usually respond right away.
But, I have no internet service right now and am troubleshooting.  I'm at a coffee shop right now.

For that transmitter, xDrip makes no modification to the readings coming from the transmitter.

I cannot do justice to your questions from a coffee shop.

I will respond properly to all your questions hopefully soon.

**Comment by Navid200** [comment]
How do you know how accurate your blood glucose meter is?  Have you compared it regularly with lab blood tests?

The logs are here: https://navid200.github.io/xDrip/docs/Logs.html

When starting a new transmitter, I recommend waiting for the battery voltages to be populated on the system status pas explained in the guide, before starting the sensor.

**Reply by Billibilli** [reply]
Could you specify how do I know the battery voltage has properly populated? Does it mean the voltage voltage A and voltage B have value?  I also got some other questions on the document. 


1) 

What are the correct routine to do in Xdrip if  replace the sensor only (still using the old transmitter)? 



2) 
[G6 Recommended Settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html)

> Disable Settings −> Less common settings −> Advanced Calibration −> Automatic Calibration

What does this do? 


3) 

Is there a conflict between _Allow OB1 unbonding_ and those controling Bluetooth on/off setting in bluetooth setting such as _G5 Bluetooth Watchdog_? If I check _G5 Bluetooth Watchdog_, does it mean I should also check _Allow OB1 unbonding_ ?


thanks for your answer.

**Reply by Navid200** [reply]
Yes, when voltage A and voltage B are shown for a new transmitter, it means you have had connectivity for a few read cycles.  And, that's when I start a sensor on a new transmitter.

The instructions apply to both starting a sensor on a new transmitter or on an existing transmitter:
https://navid200.github.io/xDrip/docs/Starting-G6.html

Automatic calibration uses blood glucose entries for calibration.  You can link a Bluetooh meter to xDrip.  Then, all your measurements are entered into xDrip as measurements.  If you enable Automatic Calibration, all those will be used for calibration if they satisfy a verification.

This is not something I recommend for a G6.

You should follow the recommended settings exactly.  There are recommended settings for Bluetooth settings, including the watchdog and there are recommended settings for the G5/G6 Debug settings.

You are free to explore.  But, I don't recommend that.

**Reply by Billibilli** [reply]
I followed the [instruction](https://navid200.github.io/xDrip/docs/Starting-G6.html) exactly  and I lost signal at night during sleep. So I started try modifying the setting, so far these are the two I changed but no luck

1) Allow OB1 unbonding (on)
2) Timer value for BT watchdog: 60min (on)

I changed them back but still no luck.  Have been fighting this for three days but no any progresses....

Waking up today morning, the signal is completely gone.... no sign of recovery

I also followed the instruction on setting the android, such as [this](https://androidaps.readthedocs.io/en/latest/Usage/huawei.html) and [this](https://androidaps.readthedocs.io/en/latest/Configuration/xdrip.html), though my phone (huawei p40) does have a different UI and management mechanism, I do the following

1) disable battery optimization,  
2) give Xdrip all possible permissions 
3) locked xdrip so it doesnt get cleaned up at background. 
4) dsiable the phone application auto management. 
5) disable the system update at night

![af6298d23b23313b9de7e507df12406](https://user-images.githubusercontent.com/4047002/187350045-7ebf5aa9-47f7-48a9-afb5-2da0360697d1.jpg)

![48de9b2ef1deddd4b0b8e8ebffb3aef1_](https://user-images.githubusercontent.com/4047002/187350256-b6d46cea-dd9e-477a-9315-226e09e1c573.jpg)


![57d98cf125fc431a4be4a2df4d24a489_](https://user-images.githubusercontent.com/4047002/187350307-9f77cc65-9ebd-41c9-b7b0-6040917721e2.jpg)

![c5f1b20a05d3253ac699a630e8199aa](https://user-images.githubusercontent.com/4047002/187352235-6e22111c-a1ae-4d70-a77b-964d0b03e367.jpg)

**Reply by Billibilli** [reply]
Is it possible to buy an iphone and reactivate the transmitter and sensor using Dexcom's app? Will there be any conflict for re-activation? 

I think probably this is Huawei's p40 fault. Xdrip is completely dead right now. But currently blood glucose is not well managed so i desperately relying on the sensor. 

Also spent a lot of money on the Dexcom sensors dont want to waste it.  There are no Dexcom selling in my local area and those are imported.

**Reply by Billibilli** [reply]
> Automatic calibration uses blood glucose entries for calibration. You can link a Bluetooh meter to xDrip. Then, all your measurements are entered into xDrip as measurements. If you enable Automatic Calibration, all those will be used for calibration if they satisfy a verification.
> 
> This is not something I recommend for a G6.
> 

I thought Automatic calibration is recommended in the documentation by you??  In the bullet point 4, it links to a page [G6 Recommended Settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html),  and you said 


> Disable Settings −> Less common settings −> Advanced Calibration −> Automatic Calibration

**Reply by Navid200** [reply]
You can stop using xDrip and use Dexcom app.  There is noting that will impeded that other than leaving xDrip active.  You cannot have both xDrip and Dexcom app collecting from the same transmitter at the same time.

The first word in the line you have copied is "Disable".  So, I am suggesting that you disable automatic calibration.
I will edit that page to put disable on a separate line to make it clear.

**Reply by Billibilli** [reply]
@Navid200 Thanks very much for your suggestions, I thought "Disable Setting" was a terminology , so silly me. I changed phone and make Xdrip works again, it is now my old Android 11 phone. So far the connection is good so I guess that was Huawei phone problem. 

But the inaccuracy is still there. I start to note the pattern is blood meter value*1.46 - 0.458 =Dexcom-Xdrip reading  .  I still trust my blood meter as it has small discrepancy with most of my old freelibre sensor readings, and was once compared with hospital lab testing. As you can see, the error model suggest that at high value the discrepancy is large enough to mistake a normal value to a dangerous one.  

![5992c4b681a4ad0d6bbf8886173a5bc](https://user-images.githubusercontent.com/4047002/187629596-e163715a-b331-4231-9361-6902206f20a3.jpg)


So I guess the error model could be y = kx +b, k and b seems consistent throughout a day, but drift slightly in other days. I searched and it seems dexcom hardware calibration has many restrictions (like the 20 rule) and don't know if it can be rolled back to raw value once the calibration go terribly wrong, and you doesn't seem to suggest any calibrations. I wonder if a software calibration is possible (like post processing the raw value it gets), and can be rolled back to raw value? or is it possible to overlay a graph with software guessed "real value" to compare against the raw value?

**Reply by Navid200** [reply]
I'm not sure why you keep referring to Libre.  What will you do when you get G7?  Will you keep tweaking it until it agrees with your Libre?  Why?

Your transmitter does not provide raw values.
Even if it did, using a G6 in non-native mode (using raw values) is like using paddles in a motorboat!  I strongly advise against that.

Dexcom G6 is a very accurate CGM.  One may need to calibrate very rarely.
However, if you believe that it is not the right CGM for you, you should discuss it with your doctor.

I would strongly be against adding post-processing calibration capability to xDrip.

**Reply by Billibilli** [reply]
Because it is a natural reasoning that if **TWO blood meters** and **ONE freelibre** say that I have a reading X, yet **ONLY ONE** Dexcom G6 says I have a reading Y, then it is very unlikely that Dexcom G6 is the right one because all these meters and CGM from different brands can hardly be wrong at the same time.  The freelibre sensor I use have been cross-compared with blood meter and the result is always consistent at fasting. And both blood meter have checked with hospital lab test two times. I know my health well. I am very sure that Dexcom G6 is the wrong one even I don't have to check with other devices, as it many times also gives me values that it is impossible on my current situation.  I wouldn't say G6 is not accurate in general but the one I tried on for the first time certainly is stressful and frustrating. But I kinda stuck with all the G6 sensors right now due to complicated reasons. I cannot return them (it is purchased out of my country) or sell them (it is illegal in my area), and my phone are not supporting the official dexcom software. So I am stuck with Xdrip and Dexcom G6 right now. 

I would like to know what could cause a persistent wrong reading from the configuration I have right now  and how can one cope under this condition. I belive there are users out there meet the same problem  and any suggestions to solve inaccuracy problem are appreciated.

**Comment by jamorham** [comment]
I don't know why this discussion is so complicated. Use the top left menu and select `Add Calibration` and enter a glucose value taken recently from a reliable blood glucose meter and this should be sent to the transmitter which should then interpret it to produce better values. Sometimes you need to do this at different times, for example first thing in the morning before any eating has occurred or if it is incorrect at a higher value try calibrating there.  Ideal times to do is when there is little change occurring because then blood glucose will be closest to interstitial which is what a CGM measures.
